lib/cim/iri-provider.ts

```diff
@@ -0,0 +1,6 @@
+export interface IriProvider {
```

Comment:
Is this really a provider? As based on the interface it is almost like an ```adapter``` or ``conversion``` method. It transforms one string to another. Also there is missing comment what this method does and why.